File: VB/TestPersistentRepository/MyPopupContainerForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports DevExpress.XtraEditors.Popup
Imports DevExpress.XtraEditors
Imports System.Drawing



Namespace DevExpress.MyControl
	Public Class MyPopupContainerForm
		Inherits PopupContainerForm
		Public Sub New(ByVal ownerEdit As PopupContainerEdit)
			MyBase.New(ownerEdit)
		End Sub




		Protected Overrides Function CalcFormSizeCore() As Size
			Dim size As Size = MyBase.CalcFormSizeCore()
			Dim editor As MyPopupContainerEdit = TryCast(OwnerEdit, MyPopupContainerEdit)
			If editor IsNot Nothing Then

				Dim s_type As String = editor.Parent.GetType().ToString()
				Dim size_obj As Object = editor.Properties.CacheSize(s_type)
				If size_obj IsNot Nothing Then
					size = CType(size_obj, Size)
				End If
			End If

			Return size
		End Function



		Protected Overrides Sub SetPropertyStore(ByVal size As Size)
			MyBase.SetPropertyStore(size)

			Dim ri As RepositoryItemMyPopupContainerEdit = TryCast(OwnerEdit.Properties, RepositoryItemMyPopupContainerEdit)

			If ri.CacheSize IsNot Nothing Then
				Dim parent_type As String = OwnerEdit.Parent.GetType().ToString()
				If (Not ri.CacheSize.ContainsKey(parent_type)) Then
					ri.CacheSize.Add(parent_type, size)
					Return
				End If

				ri.CacheSize(parent_type) = size
			End If
		End Sub
	End Class
End Namespace

File: CS/TestPersistentRepository/MyPopupContainerForm.cs
using System;
using System.Collections.Generic;
using DevExpress.XtraEditors.Popup;
using DevExpress.XtraEditors;
using System.Drawing;



namespace DevExpress.MyControl
{
    public class MyPopupContainerForm : PopupContainerForm
    {
        public MyPopupContainerForm(PopupContainerEdit ownerEdit)
            : base(ownerEdit)
        {
		}



        protected override Size CalcFormSizeCore()
        {
            Size size = base.CalcFormSizeCore();
            MyPopupContainerEdit editor = OwnerEdit as MyPopupContainerEdit;
            if (editor != null)
            {

                string s_type = editor.Parent.GetType().ToString();
                object size_obj = editor.Properties.CacheSize[s_type];
                if (size_obj != null) size = (Size)size_obj;
            }

            return size;
        }



        protected override void SetPropertyStore(Size size)
        {
            base.SetPropertyStore(size);

            RepositoryItemMyPopupContainerEdit ri = OwnerEdit.Properties as RepositoryItemMyPopupContainerEdit;
            
            if (ri.CacheSize != null)
            {
                string parent_type = OwnerEdit.Parent.GetType().ToString();
                if (!ri.CacheSize.ContainsKey(parent_type))
                {
                    ri.CacheSize.Add(parent_type, size);
                    return;
                }

                ri.CacheSize[parent_type] = size;
            }
        }
    }
}
